src/System.IO.FileSystem/tests/PathFeatures.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Reflection;
+using System.Runtime.InteropServices;
+
+namespace System.IO
+{
+    public static class PathFeatures
+    {
+        // Note that this class is using APIs that allow it to run on all platforms (including Core 5.0)
+        // That is why we have .GetTypeInfo(), don't use the Registry, etc...
+
+        private static bool? s_osEnabled;
+        private static bool? s_onCore;
+
+        /// <summary>
+        /// Returns true if you can use long paths, including long DOS style paths (e.g. over 260 without \\?\).
+        /// </summary>
+        public static bool AreAllLongPathsAvailable()
+        {
+            // We have support built-in for all platforms in Core
+            if (RunningOnCore)
+                return true;
+
+            // Otherwise we're running on Windows, see if we've got the capability in .NET, and that the feature is enabled in the OS
+            return !AreLongPathsBlocked() && AreOsLongPathsEnabled();
+        }
+
+        public static bool IsUsingLegacyPathNormalization()
+        {
+            return HasLegacyIoBehavior("UseLegacyPathHandling");
+        }
+
+        /// <summary>
+        /// Returns true if > MAX_PATH (260) character paths are blocked.
+        /// Note that this doesn't reflect that you can actually use long paths without device syntax when on Windows.
+        /// Use AreAllLongPathsAvailable() to see that you can use long DOS style paths if on Windows.
+        /// </summary>
+        public static bool AreLongPathsBlocked()
+        {
+            return HasLegacyIoBehavior("BlockLongPaths");
+        }
+
+        private static bool HasLegacyIoBehavior(string propertyName)
+        {
+            // Core doesn't have legacy behaviors
+            if (RunningOnCore)
+                return false;
+
+            Type t = typeof(object).GetTypeInfo().Assembly.GetType("System.AppContextSwitches");
+            var p = t.GetProperty(propertyName, BindingFlags.Static | BindingFlags.Public);
+
+            // If the switch actually exists use it, otherwise we predate the switch and are effectively on
+            return (bool)(p?.GetValue(null) ?? true);
+        }
+
+        private static bool RunningOnCore
```

Comment:
I think we need a better way in general to detect that. System.Private.CoreLib exists in UAP too. so your check will include such platform. this maybe ok in your case.